portal-web/docroot/html/taglib/ui/search_iterator/lexicon/bottom.jspf

```diff
@@ -94,39 +94,112 @@
 
 		<c:if test="<%= fixedHeader && displayStyle.equals(SearchIteratorTag.DEFAULT_DISPLAY_STYLE) %>">
 			var contentBox = searchContainer.get('contentBox');
-
-			var fixedHeader = document.getElementById('<%= namespace + id %>fixedHeader');
+			var insideIframe = window !== Liferay.Util.getOpener();
 
 			if (contentBox) {
+				var tableElement;
+
+				var fixedHeader = document.getElementById('<%= namespace + id %>fixedHeader');
 				var mainContent = contentBox.ancestor('#main-content');
+				var tableResponsive = contentBox.one('.table-responsive');
 
-				if (mainContent) {
-					var scrollingContainer = window === Liferay.Util.getOpener() ? window : mainContent._node;
+				var registerEventListener = function(element, eventType, eventListener) {
+					element.addEventListener(eventType, eventListener);
+
+					Liferay.on(
+						'destroyPortlet',
+						function() {
+							element.removeEventListener(eventType, eventListener);
+						}
+					);
+				};
 
-					var table = fixedHeader.parentElement.parentElement;
+				if (mainContent) {
+					var scrollingContainer = insideIframe ? mainContent._node : window;
 					var trDomRect = fixedHeader.previousElementSibling.getBoundingClientRect();
 
-					var fixedHeaderEventHandler = function(event) {
-						var scrollTop = scrollingContainer.scrollTop;
+					tableElement = fixedHeader.parentElement.parentElement;
 
-						var trDomRecTop = trDomRect.top;
+					registerEventListener(
+						scrollingContainer,
+						'scroll',
+						function(event) {
+							var scrollTop = scrollingContainer.scrollTop;
 
-						fixedHeader.style.transform = 'translateY(' + Math.ceil(scrollTop - table.offsetTop - trDomRect.height) + 'px)';
+							var trDomRecTop = trDomRect.top;
 
-						if (fixedHeader.classList.contains('hide') && (scrollTop >= trDomRecTop)) {
-							fixedHeader.classList.remove('hide');
+							fixedHeader.style.transform = 'translateY(' + Math.ceil(scrollTop - tableElement.offsetTop - trDomRect.height) + 'px)';
+
+							if (fixedHeader.classList.contains('hide') && (scrollTop >= trDomRecTop)) {
+								fixedHeader.classList.remove('hide');
+							}
+							else if (scrollingContainer.scrollTop < trDomRecTop) {
+								fixedHeader.classList.add('hide');
+							}
 						}
-						else if (scrollingContainer.scrollTop < trDomRecTop) {
-							fixedHeader.classList.add('hide');
+					);
+
+					if (insideIframe) {
+						scrollingContainer.querySelector('.button-holder').classList.add('lfr-search-iterator-fixed-header-button-holder');
```

Comment:
We should check to make sure the `button-holder` CSS classes exists and DOM node is returned before using in JS.